src/test/java/com/example/demo/SecurityIntegrationTest.java

```diff
@@ -0,0 +1,178 @@
+                .andReturn();
+        assertThat(result.getResponse().getStatus()).isEqualTo(HttpStatus.BAD_REQUEST.value());
+    }
+
+    @Test
+    public void itShouldNotAllowLoginWithBadCredentials() throws Exception {
+        String requestBody = TestUtils.asJsonString(new AuthRequest("badUsername", "badPassword"));
+
+        MvcResult result = mvc.perform(
+                        MockMvcRequestBuilders.post("/auth/login")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(requestBody))
+                .andReturn();
+        assertThat(result.getResponse().getStatus()).isEqualTo(HttpStatus.UNAUTHORIZED.value());
+    }
+
+    @Test
+    public void itShouldNotAllowPrivateAccessWithoutToken() throws Exception {
+        String testUser1Accountid = "GET THIS ONCE USERS HAVE ACCOUNTIDS";
+        MvcResult result = mvc.perform(
+                        MockMvcRequestBuilders.get("/orderbook/buy/" + testUser1Accountid))
+                .andReturn();
+        assertThat(result.getResponse().getStatus()).isEqualTo(HttpStatus.UNAUTHORIZED.value());
+    }
+
+    @Test
+    public void itShouldNotAllowPrivateAccessWithBadlyFormattedToken() throws Exception {
+        String testUser1Accountid = "GET THIS ONCE USERS HAVE ACCOUNTIDS";
+        MvcResult result = mvc.perform(
+                        MockMvcRequestBuilders.get("/orderbook/buy/" + testUser1Accountid)
+                                .header("Authorization", "Bearer badtoken"))
+                .andReturn();
+        assertThat(result.getResponse().getStatus()).isEqualTo(HttpStatus.UNAUTHORIZED.value());
+    }
+
+    @Test
+    public void itShouldNotAllowPrivateAccessWithOldToken() throws Exception {
+        String testUser1Accountid = "GET THIS ONCE USERS HAVE ACCOUNTIDS";
+        MvcResult result = mvc.perform(
+                        MockMvcRequestBuilders.get("/orderbook/buy/" + testUser1Accountid)
+                                .header("Authorization", "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9 .eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ .SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c"))
+                .andReturn();
+        assertThat(result.getResponse().getStatus()).isEqualTo(HttpStatus.UNAUTHORIZED.value());
+    }
+
+    @Test
+    public void itShouldAllowAccessToPublicBuyOrderbook() throws Exception {
+        MvcResult result = mvc.perform(
+                        MockMvcRequestBuilders.get("/orderbook/buy"))
+                .andReturn();
+        assertThat(result.getResponse().getStatus()).isEqualTo(HttpStatus.OK.value());
+    }
+
+    @Test
+    public void itShouldAllowAccessToPublicSellOrderbook() throws Exception {
+        MvcResult result = mvc.perform(
+                        MockMvcRequestBuilders.get("/orderbook/sell"))
+                .andReturn();
+        assertThat(result.getResponse().getStatus()).isEqualTo(HttpStatus.OK.value());
```

Comment:
Again in here. I'm not sure if we need to do these tests again if you already check for `HttpStatus.ok.value()` in other tests. My suggestion would be to get rid of these from here. I'm happy for this file to just test the bad requests as you already tested the "happy paths" in the MatcherController tests.

Main reason for not duplicating tests is, if for some reason that test needs to be updated you'll need to update it in more than one place and it's not something we want to do. 

Hope that makes sense let me know if you have any questions about this!